android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageViewModel.kt

```diff
@@ -28,12 +29,30 @@ class ListingDetailsPageViewModel @Inject constructor(
 	private val locationService: ILocationService,
 ) : SubletrViewModel<ListingDetailsPageUiState>() {
 	val navHostController: NavHostController get() = navigationService.navHostController
+	private val listingId: Int = checkNotNull(savedStateHandle["listingId"])
 
-	private val listingIdStream: BehaviorSubject<Int> =
-		BehaviorSubject.createDefault(checkNotNull(savedStateHandle["listingId"]))
+	private val listingIdStream: BehaviorSubject<Int> = BehaviorSubject.createDefault(listingId)
 	private val imageIdsStream: BehaviorSubject<List<String>> =
 		BehaviorSubject.createDefault(emptyList())
-	val favouritedStream: BehaviorSubject<Boolean> = BehaviorSubject.createDefault(false)
+	private val favouritedStream: BehaviorSubject<Boolean> = BehaviorSubject.createDefault(false)
```

Comment:
Should probably be a PublishSubject.
BehaviourSubject are for when you want it to remember and re-emit state.

Reply:
`toggleFavouriteStream` is the `PublishSubject` instead